### PR TITLE
Subform table

### PR DESCRIPTION
**Pull request by neb42**
* Adds a table to render subform items
* Optional `onDelete` for deleting them when in edit mode
* Added a `naked` button variant for the delete icon in the table (needs work)
* Icon alignment in buttons seems broken

### CODE REVIEW

File: frontend/apps/core-app/src/components/RecordView/normaliseFieldValues.ts [hunk 1]
```diff
@@ -24,39 +23,33 @@ export const normaliseFieldValues = (
...
             const subField = field.fieldType.subForm?.fields.find(
-              (f) => f.id === subFieldValue.fieldId,
+              (f) => f.id === c.fieldId,
             );
             if (!subField) throw new Error('subField not found');
-            const subFieldType = getFieldKind(subField.fieldType);
-            if (subFieldType === FieldKind.SubForm)
-              throw new Error('subField is a subform');
+            if (getFieldKind(subField.fieldType) === FieldKind.SubForm)
+              throw new Error('Nested subforms are not supported');
```
Comment (internetti) [inline frontend/apps/core-app/src/components/RecordView/normaliseFieldValues.ts:43]: also not by the api, or just not handled in the frontend?

File: frontend/apps/core-app/src/components/SubFormTable/SubFormTable.component.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      <VStack width="100%">
+        <HStack width="100%">
+          {table.columns.map((col) => (
+            <Box
+              key={col.id}
+              width={col.width}
+              p="2"
+              borderBottomColor="neutral.300"
+              borderBottomWidth="1"
+              // flexGrow={1}
```
Comment (internetti) [inline frontend/apps/core-app/src/components/SubFormTable/SubFormTable.component.tsx:25]: WIP or can this go?

File: frontend/apps/core-app/src/components/SubFormTable/SubFormTable.component.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  return (
+    <ScrollView width="100%" horizontal>
+      <VStack width="100%">
+        <HStack width="100%">
+          {table.columns.map((col) => (
+            <Box
+              key={col.id}
+              width={col.width}
+              p="2"
+              borderBottomColor="neutral.300"
```
Comment (internetti) [inline frontend/apps/core-app/src/components/SubFormTable/SubFormTable.component.tsx:23]: not a big issue, and design might change anyway, but the bottom border on the table header is shorter due to the delete buttons, looks weird

File: frontend/apps/core-app/src/utils/ReactHookFormTransformer.ts [hunk 4]
```diff
@@ -18,16 +41,50 @@ export const toReactHookForm = (
...
+          const v = value[datum.form.id][field.id].map(
+            (vv) =>
+              fromReactHookForm(
+                [
+                  {
+                    form: {
+                      id: field.id,
+                      databaseId: datum.form.databaseId,
+                      folderId: datum.form.folderId,
+                      name: '',
```
Comment (internetti) [inline frontend/apps/core-app/src/utils/ReactHookFormTransformer.ts:62]: do we not need a name here? isn't formname required?

File: frontend/apps/core-app/src/components/RecordView/normaliseFieldValues.ts (hunk 1 above)
Comment (neb42) [reply frontend/apps/core-app/src/components/RecordView/normaliseFieldValues.ts:43]: I think the discussion concluded on, the api does support it but we should remove it at some point. The pwa does not support adding them.

File: frontend/apps/core-app/src/utils/ReactHookFormTransformer.ts (hunk 4 above)
Comment (neb42) [reply frontend/apps/core-app/src/utils/ReactHookFormTransformer.ts:62]: It's required, but not used by subforms

**Pull request merged**